agent/jvm/src/main/java/org/jolokia/jvmagent/JolokiaServer.java

```diff
@@ -149,7 +149,11 @@ public void stop() {
         serviceManager.stop();
 
         if (cleaner != null) {
-            cleaner.stopServer();
+            try {
+                cleaner.stopServer();
+            } catch (InterruptedException exp) {
+                System.err.println("Could not stop Jolokia server properly: " + exp);
```

Comment:
I wouldnt print to stderr here, since JolokiaServer is also used by other directly, so its not sure in which circumstance its running. Instead the `LogHandler` should be used because it can be muted, too. (Add a member variable `log` in `init()` for this)